acceptancetests/assess_caas_deploy_charms.py

```diff
@@ -12,213 +12,73 @@
+    if not callable(success_hook) or not callable(fail_hook):
+        raise RuntimeError("hooks are not callable")
 
-def check_app_healthy(url, timeout=300):
     status_code = None
-    for _ in until_timeout(timeout):
+    for remaining in until_timeout(timeout):
         try:
             r = requests.get(url)
             if r.ok and r.status_code < 400:
-                return
+                return success_hook()
             status_code = r.status_code
         except IOError as e:
             log.error(e)
-        sleep(3)
+        finally:
+            sleep(3)
+            if remaining % 60 == 0:
+                log.info('timeout in %ss', remaining)
     log.error('HTTP health check failed -> %s, status_code -> %s !', url, status_code)
+    fail_hook()
     raise JujuAssertionError('gitlab is not healthy')
 
 
-def assess_caas_charm_deployment(client):
-    # Deploy k8s bundle to spin up k8s cluster
-    bundle = local_charm_path(
-        charm='bundles-kubernetes-core-lxd.yaml',
-        repository=os.environ['JUJU_REPOSITORY'],
-        juju_ver=client.version
-    )
-
-    caas_client = deploy_caas_stack(bundle_path=bundle, client=client)
+def assess_caas_charm_deployment(caas_client):
     external_hostname = caas_client.get_external_hostname()
 
-    if not caas_client.is_cluster_healthy:
+    if not caas_client.check_cluster_healthy(timeout=60):
         raise JujuAssertionError('k8s cluster is not healthy because kubectl is not accessible')
 
-    # tmp fix kubernetes core ingress issue
-    caas_client.kubectl(
-        'patch', 'daemonset.apps/nginx-ingress-kubernetes-worker-controller', '--patch',
-        '''
-        {"spec": {"template": {"spec": {"containers": [{"name": "nginx-ingress-kubernetes-worker","args": ["/nginx-ingress-controller", "--default-backend-service=$(POD_NAMESPACE)/default-http-backend", "--configmap=$(POD_NAMESPACE)/nginx-load-balancer-conf", "--enable-ssl-chain-completion=False", "--publish-status-address=%s"]}]}}}}
-        ''' % caas_client.get_first_worker_ip()
-    )
-
     # add caas model for deploying caas charms on top of it
     model_name = 'testcaas'
     k8s_model = caas_client.add_model(model_name)
 
-    # ensure tmp dir for storage class.model_name
-    o = subprocess.check_output(
-        ('sudo', 'mkdir', '-p', '/mnt/kubernetes/%s' % model_name)  # unfortunately, needs sudo
-    )
-    log.debug(o.decode('UTF-8').strip())
-
-    # ensure storage class
-    caas_client.kubectl_apply(HOST_PATH_PROVISIONER.format(class_name=JUJU_STORAGECLASS_NAME))
-
-    # ensure storage pools for caas operator
+    sc_name = caas_client.default_storage_class_name
+    # ensure storage pools for caas operator using default sc.
     k8s_model.juju(
         'create-storage-pool',
-        ('operator-storage', 'kubernetes', 'storage-class=%s' % JUJU_STORAGECLASS_NAME)
+        ('operator-storage', 'kubernetes', 'storage-class=%s' % sc_name)
```

Comment:
We don't need to create storage pools by default anymore.
Please add a todo to deploy without, but there should be a separate test which does add a storage pool

Reply:
done.